lib/jekyll/filters/url_filters.rb

```diff
@@ -12,6 +12,7 @@ module URLFilters
       # Returns the absolute URL as a String.
       def absolute_url(input)
         return if input.nil?
+        input = input.url if input.respond_to?(:url)
         return input if Addressable::URI.parse(input.to_s).absolute?
         site = @context.registers[:site]
         return relative_url(input) if site.config["url"].nil?
```

Comment:
in the case where `site.config["url"].nil?`, I think the `respond_to?(:url)` will be called again and therefore `input.url` will be called once again.. and consequently, a **re-assignment** of `input` unnecessarily